compiler/rustc_span/src/lib.rs

```diff
@@ -1886,6 +1886,50 @@ macro_rules! impl_pos {
             $(#[$attr])*
             $vis struct $ident($inner_vis $inner_ty);
 
+            impl ::std::cmp::Ord for $ident {
+                #[inline(always)]
+                fn cmp(&self, other: &Self) -> ::std::cmp::Ordering {
+                    self.0.cmp(&other.0)
+                }
+
+                #[inline(always)]
+                fn min(self, other: Self) -> Self {
+                    Self(self.0.min(other.0))
+                }
+
+                #[inline(always)]
+                fn max(self, other: Self) -> Self {
+                    Self(self.0.max(other.0))
+                }
+            }
+
+            impl ::std::cmp::PartialOrd for $ident {
```

Comment:
suggestion: if this ends up still being needed, I think it'd be good to have a `FIXME` comment to move it back to being derived once that's no longer a perf regression.

(Although I'm worried that if it *is* still needed, then everyone else's newtypes like this might also be regressed, which seems pretty scary, and a reason it might not want to land at all.)

Reply:
I agree that if this is required it is reason to not land this. I'm only leaving this in for now because I want to isolate effects as much as possible.